storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -90,6 +92,23 @@ public Cluster(INimbus nimbus, Map<String, SupervisorDetails> supervisors, Map<S
         }
         this.conf = storm_conf;
     }
+
+    /**
+     * Get a copy of this cluster object
+     */
+    public Cluster getCopy() {
+        HashMap<String, SchedulerAssignmentImpl> newAssignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignmentImpl> entry : this.assignments.entrySet()) {
+            newAssignments.put(entry.getKey(), new SchedulerAssignmentImpl(entry.getValue().getTopologyId(), entry.getValue().getExecutorToSlot()));
+        }
+        Map newConf = new HashMap<String, Object>();
+        newConf.putAll(this.conf);
+        Cluster copy = new Cluster(this.inimbus, this.supervisors, newAssignments, newConf);
+        for (Map.Entry<String, String> entry : this.status.entrySet()) {
+            copy.setStatus(entry.getKey(), entry.getValue());
+        }
```

Comment:
The for loop is OK but you could just `copy.status = new HashMap<>(this.status);`.  (very minor)